In [1]:
import math

In [2]:
n = 17513
a = 10

This is my Pollard Function

it takes in the a value from above and the composite number we need to factor

it outputs a factor of the composite number

It works by looping from 2..9 (2,3,4,5,6,7,8,9)

- find b in the formula b = a^(k!) where a is provided and k is generated during the loop

- then gets the gcd(b-1, n) where b is calculated above and n is the composite number we want to factor

- stop checking when the gcd is not 1

- returns the gcd as that is one of the factors

In [3]:
def pollard(a, composite):
    for k in range(2, 10):
        b = a ** math.factorial(k)
        b -= 1
        b %= composite
        res = math.gcd(int(b), composite)
        if(res != 1):
            break
        else:
            print("not", k)
            
    return res

In [4]:
compositeNum = n
pollardResult = pollard(a, compositeNum)
print(pollardResult)

not 2
not 3
not 4
211


now I know that 211 is one of the factors of 17513
now I divide 17513 by 211 and check if its prime

In [5]:
potentialPrime = compositeNum // pollardResult
for j in range(2, potentialPrime // 2 + 1):
    if(potentialPrime % j == 0):
        break; 
else:
    print(potentialPrime ,"is a Prime!")

83 is a Prime!


with this Both of my numbers are prime and there are no more composite numbers to check so the factors have to be 211 and 83

>Final Ans: 211 * 83